### PR TITLE
[CORE-2227] Allocate a main context in pachctl main

### PR DESCRIPTION
**Pull request by robert-uhl**
And plumb through to `PachctlCmd` and command execution.

### CODE REVIEW

File: src/server/cmd/pachctl/cmd/cmd_test.go [hunk 1]
```diff
@@ -37,8 +38,9 @@ func TestPortForwardError(t *testing.T) {
 // 'CreateAlias' was not used properly (or the command just needs to specify
 // its name).
 func TestCommandAliases(t *testing.T) {
+	ctx := pctx.TestContext(t)
 	log.Test(context.Background(), t) // Just to capture global log messages; prefer pctx.TestContext in newer code.
```
Comment (jrockway) [inline src/server/cmd/pachctl/cmd/cmd_test.go:42]: log.Test can be removed now.

File: src/server/cmd/pachctl/main.go [hunk 2]
```diff
@@ -7,12 +7,14 @@ import (
...
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
 	"github.com/pachyderm/pachyderm/v2/src/internal/tracing"
 	"github.com/pachyderm/pachyderm/v2/src/server/cmd/pachctl/cmd"
 	"github.com/spf13/pflag"
 )
 
 func main() {
+	ctx := pctx.Background("pachctl")
```
Comment (jrockway) [inline src/server/cmd/pachctl/main.go:17]: This needs to happen after InitPachctlLogger.

**Review by jrockway — APPROVED** [review]
I am not sure this makes things better without a refactoring to remove mainCtx from the command builders and a changeover from mainCtx to cmd.Context.  People still won't know which to use for new RPCs. 

Also now this blocks other pachctl work until we decide what state we want to leave this in, where before the PR, we just did things in an unusual but consistent way.

File: src/server/cmd/pachctl/main.go (hunk 2 above)
Comment (robert-uhl) [reply src/server/cmd/pachctl/main.go:17]: It’s a bit surprising to combine global and dynamic state — would it be preferable to initialize the logger inside `pctx.Background`, or return a pachtl-specific context from `pctx.InitPachctlLogger`, or have `pachctl` mutate its contextual logger?

Regardless, update to move it afterwards.

**Comment by robert-uhl** [comment]
> I am not sure this makes things better without a refactoring to remove mainCtx from the command builders and a changeover from mainCtx to cmd.Context. People still won't know which to use for new RPCs.

It may not make things better, but it doesn’t make them worse, right? Before this PR, `main` called `PachctlCmd`, which allocated a `pctx` background context, which passed it to command-handler-creators, and some command handlers closed over that context and others didn’t, and then the selected command handler was called with a new `context.Context`; after this PR, `main` allocates a `pctx` background context, passes it to `PachctlCmd`, which passes it to command-handler-creators, and some command handlers close over that context and other don’t, and then the selected command handler is called with that same initial `pctx` background context.

> Also now this blocks other pachctl work until we decide what state we want to leave this in, where before the PR, we just did things in an unusual but consistent way.

_Does_ this block any work? I believe that it is a strict improvement and doesn’t leave anything more broken than it was before. But … finding out that one’s PR will break something one had no idea about is definitely a key reason for PRs☺

File: src/server/cmd/pachctl/cmd/cmd.go [hunk 3]
```diff
@@ -868,20 +865,20 @@ This resets the cluster to its initial state.`,
...
-	subcommands = append(subcommands, identitycmds.Cmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, admincmds.Cmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, debugcmds.Cmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, txncmds.Cmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, configcmds.Cmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, configcmds.ConnectCmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, taskcmds.Cmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, misccmds.Cmds(mainCtx, pachctlCfg)...)
-	subcommands = append(subcommands, metadatacmds.Cmds(mainCtx, pachctlCfg)...)
+	subcommands = append(subcommands, pfscmds.Cmds(pachCtx, pachctlCfg)...)
```
Comment (jrockway) [inline src/server/cmd/pachctl/cmd/cmd.go:868]: I think this is the place where you should get rid of the context parameter and require the commands created in there to get it out of cmd.Context()

**Review by FahadBSyed — APPROVED** [review]
Thumbs up from PFS 👍

**Pull request merged**